lib/rules/no-unused-prop-types.js

```diff
@@ -104,7 +104,7 @@ module.exports = {
 
         if (prop.node && !isPropUsed(component, prop)) {
           context.report({
-            node: prop.node.value || prop.node,
+            node: prop.node.key || prop.node,
```

Comment:
I wonder if the fallback to `prop.node` here is actually needed? If the fallback is used then it will report like this:

<img width="296" alt="Screen Shot 2019-05-27 at 08 02 12" src="https://user-images.githubusercontent.com/9586897/58398574-c52ac200-8055-11e9-97fc-87a3a21d3bc4.png">

Reply:
Maybe it is for flow object type? It seems that babel-eslint does not provide `key` in `ObjectTypeProperty` nodes [demo](https://astexplorer.net/#/gist/6e58a5503fe6641226fa05d744793698/bac2bea3e2ca012ba38e5950646a7d69648587f4).